x/bank/internal/keeper/keeper.go

```diff
@@ -223,119 +238,23 @@ func (keeper BaseSendKeeper) SendCoins(
-// account balances.
-type ViewKeeper interface {
-	GetCoins(ctx sdk.Context, addr sdk.AccAddress) sdk.Coins
-	HasCoins(ctx sdk.Context, addr sdk.AccAddress, amt sdk.Coins) bool
-
-	Codespace() sdk.CodespaceType
-}
-
-// BaseViewKeeper implements a read only keeper implementation of ViewKeeper.
-type BaseViewKeeper struct {
-	ak        types.AccountKeeper
-	codespace sdk.CodespaceType
-}
-
-// NewBaseViewKeeper returns a new BaseViewKeeper.
-func NewBaseViewKeeper(ak types.AccountKeeper, codespace sdk.CodespaceType) BaseViewKeeper {
-	return BaseViewKeeper{ak: ak, codespace: codespace}
-}
-
-// Logger returns a module-specific logger.
-func (keeper BaseViewKeeper) Logger(ctx sdk.Context) log.Logger {
-	return ctx.Logger().With("module", fmt.Sprintf("x/%s", types.ModuleName))
-}
-
-// GetCoins returns the coins at the addr.
-func (keeper BaseViewKeeper) GetCoins(ctx sdk.Context, addr sdk.AccAddress) sdk.Coins {
-	return getCoins(ctx, keeper.ak, addr)
-}
-
-// HasCoins returns whether or not an account has at least amt coins.
-func (keeper BaseViewKeeper) HasCoins(ctx sdk.Context, addr sdk.AccAddress, amt sdk.Coins) bool {
-	return hasCoins(ctx, keeper.ak, addr, amt)
-}
-
-// Codespace returns the keeper's codespace.
-func (keeper BaseViewKeeper) Codespace() sdk.CodespaceType {
-	return keeper.codespace
-}
-
-// getCoins returns the account coins
-func getCoins(ctx sdk.Context, am types.AccountKeeper, addr sdk.AccAddress) sdk.Coins {
-	acc := am.GetAccount(ctx, addr)
-	if acc == nil {
-		return sdk.NewCoins()
-	}
-	return acc.GetCoins()
-}
-
-func setCoins(ctx sdk.Context, ak types.AccountKeeper, addr sdk.AccAddress, amt sdk.Coins) sdk.Error {
-	if !amt.IsValid() {
-		return sdk.ErrInvalidCoins(amt.String())
-	}
-
-	acc := ak.GetAccount(ctx, addr)
-	if acc == nil {
-		acc = ak.NewAccountWithAddress(ctx, addr)
-	}
-
-	err := acc.SetCoins(amt)
-	if err != nil {
-		// Handle w/ #870
-		panic(err)
-	}
-
-	ak.SetAccount(ctx, acc)
 	return nil
 }
 
-// HasCoins returns whether or not an account has at least amt coins.
-func hasCoins(ctx sdk.Context, ak types.AccountKeeper, addr sdk.AccAddress, amt sdk.Coins) bool {
-	return getCoins(ctx, ak, addr).IsAllGTE(amt)
-}
-
-// getAccount implements AccountKeeper
-func getAccount(ctx sdk.Context, ak types.AccountKeeper, addr sdk.AccAddress) exported.Account {
-	return ak.GetAccount(ctx, addr)
-}
-
-// setAccount implements AccountKeeper
-func setAccount(ctx sdk.Context, ak types.AccountKeeper, acc exported.Account) {
-	ak.SetAccount(ctx, acc)
-}
-
-// subtractCoins subtracts amt coins from an account with the given address addr.
+// SubtractCoins subtracts amt from the coins at the addr.
 //
 // CONTRACT: If the account is a vesting account, the amount has to be spendable.
-func subtractCoins(ctx sdk.Context, ak types.AccountKeeper, addr sdk.AccAddress, amt sdk.Coins) (sdk.Coins, sdk.Error) {
+func (keeper BaseSendKeeper) SubtractCoins(
```

Comment:
The function signature wasn't verbose enough to warrant a restructuring of format (i.e. I can still view the diff side-by-side fine. Not blocking, just a style preference.